client/my-sites/email/form/mailboxes/index.ts

```diff
@@ -4,49 +4,82 @@ import {
 	FIELD_FIRSTNAME,
 	FIELD_LASTNAME,
 	FIELD_MAILBOX,
+	FIELD_NAME,
 	FIELD_PASSWORD,
 	FIELD_UUID,
 } from 'calypso/my-sites/email/form/mailboxes/constants';
 import {
 	EmailProvider,
+	MailboxFormFieldBase,
 	MailboxFormFieldsFactory,
 } from 'calypso/my-sites/email/form/mailboxes/types';
 import {
 	AlternateEmailValidator,
-	ExistingMailboxesValidator,
+	ExistingMailboxNamesValidator,
 	MailboxNameValidator,
 	PasswordValidator,
 	RequiredValidator,
+	RequiredIfVisibleValidator,
+	MaximumStringLengthValidator,
 } from 'calypso/my-sites/email/form/mailboxes/validators';
 import type {
-	ValidatorFieldNames,
+	FormFieldNames,
 	MailboxFormFields,
+	ValidatorFieldNames,
 } from 'calypso/my-sites/email/form/mailboxes/types';
 import type { Validator } from 'calypso/my-sites/email/form/mailboxes/validators';
 
 class MailboxForm< T extends EmailProvider > {
-	existingMailboxes: string[];
+	existingMailboxNames: string[];
 	formFields: MailboxFormFields;
 	provider: T;
-	validators: [ ValidatorFieldNames, Validator< unknown > ][] = [
-		[ FIELD_ALTERNATIVE_EMAIL, new AlternateEmailValidator< T >( this ) ],
-		[ FIELD_DOMAIN, new RequiredValidator< string >() ],
-		[ FIELD_FIRSTNAME, new RequiredValidator< string >() ],
-		[ FIELD_LASTNAME, new RequiredValidator< string >() ],
-		[ FIELD_MAILBOX, new RequiredValidator< string >() ],
-		[ FIELD_MAILBOX, new ExistingMailboxesValidator( this ) ],
-		[ FIELD_MAILBOX, new MailboxNameValidator< T >( this ) ],
-		[ FIELD_PASSWORD, new RequiredValidator< string >() ],
-		[ FIELD_PASSWORD, new PasswordValidator() ],
-		[ FIELD_UUID, new RequiredValidator< string >() ],
-	];
-
-	constructor( provider: T, domain: string, existingMailboxes: string[] = [] ) {
-		this.existingMailboxes = existingMailboxes;
+
+	constructor( provider: T, domain: string, existingMailboxNames: string[] = [] ) {
+		this.existingMailboxNames = existingMailboxNames;
 		this.formFields = MailboxFormFieldsFactory.create( provider, domain );
 		this.provider = provider;
 	}
 
+	private getFormField< T >( fieldName: FormFieldNames ): MailboxFormFieldBase< T > | null {
+		if ( fieldName in this.formFields ) {
+			const field = Reflect.get( this.formFields, fieldName );
+			if ( field ) {
+				return field;
+			}
+		}
+
+		return null;
+	}
+
+	private getValidators(): [ ValidatorFieldNames, Validator< unknown > ][] {
+		const domainField = this.getFormField< string >( FIELD_DOMAIN );
+		const domainName = domainField?.value ?? '';
+		const mailboxHasDomainError = Boolean( domainField?.error );
+		const minimumPasswordLength = this.provider === EmailProvider.Titan ? 10 : 12;
+		const supportsApostrophes = this.provider === EmailProvider.Google;
```

Comment:
Not mandatory at all but I would suggest renaming `supportsApostrophes` to `areApostrophesSupported` (or `areApostrophesAllowed`) to be more consistent with how we usually name boolean variables.